torchgeo/datasets/seco.py

```diff
@@ -0,0 +1,233 @@
+                self.scene_to_patches[scene_name].append(patch_name)
+
+        self.scenes = sorted(self.scene_to_patches.keys())
+        for scene_name in self.scenes:
+            self.scene_to_patches[scene_name] = sorted(
+                self.scene_to_patches[scene_name]
+            )
+
+    def __getitem__(self, index: int) -> Dict[str, Tensor]:
+        """Return an index within the dataset.
+
+        Args:
+            index: index to return
+
+        Returns:
+            sample with an "image" in 5xCxHxW format where the 5 indexes over the same
+                patch sampled from different points in time by the SeCo method
+        """
+        scene_name = self.scenes[index]
+        patch_names = self.scene_to_patches[scene_name]
+
+        imagery = [
+            self._load_patch(scene_name, patch_name) for patch_name in patch_names
+        ]
+
+        sample = {"image": torch.stack(imagery, dim=0)}
+
+        if self.transforms is not None:
+            sample = self.transforms(sample)
+
+        return sample
+
+    def __len__(self) -> int:
+        """Return the number of data points in the dataset.
+
+        Returns:
+            length of the dataset
+        """
+        return len(self.scenes)
+
+    def _load_patch(self, scene_name: str, patch_name: str) -> Tensor:
+        """Load a single image patch.
+
+        Args:
+            scene_name: the name of the scene to load from, e.g. '019999'
+            patch_name: the name of the patch to load, e.g.
+                '20200713T075609_20200713T081050_T36QZH'
+
+        Returns:
+            the image with the subset of bands specified by ``self.bands``
+        """
+        all_data = []
+        for band in self.bands:
+            fn = os.path.join(
+                self.root,
+                self.directory_name,
+                scene_name,
+                patch_name,
+                f"{band}.tif",
+            )
+            with rasterio.open(fn) as f:
+                band_data = f.read(1)
+                height, width = band_data.shape
+                assert height == width
+                if height < 264 and width < 264:
+                    # TODO: PIL resize is much slower than cv2, we should check to see
+                    # what could be sped up throughout later. There is also a potential
+                    # slowdown here from converting to/from a PIL Image just to resize.
+                    # https://gist.github.com/calebrob6/748045ac8d844154067b2eefa47de92f
+                    pil_image = Image.fromarray(band_data)
+                    band_data = np.array(
+                        pil_image.resize((264, 264), resample=Image.BILINEAR)
+                    )
+                all_data.append(band_data)
+        image = torch.from_numpy(  # type: ignore[attr-defined]
+            np.stack(all_data, axis=0)
+        )
+        return cast(Tensor, image)
+
+    def _check_integrity(self) -> bool:
```

Comment:
This has the same problem as our older datasets where you can't delete the zipfile afterwards if you want to use the dataset. At some point we need to convert the remaining datasets to the new format, but since this dataset hasn't been merged yet I would prefer to use the new format. See CDL for an example of what this should look like.

Reply:
totally, good catch